src/mapping/impl/BasisFunctions.hpp

```diff
@@ -1,5 +1,25 @@
 #pragma once
 
+#ifdef __NVCC__
+
+#include <cuda_runtime.h>
+#define BOOST_PP_VARIADICS 1
+#define PRECICE_HOST_DEVICE __host__ __device__
+#define PRECICE_MEMORY_SPACE __device__
```

Comment:
```suggestion
#define PRECICE_DEVICE_MEMORY __device__
```
the memory space itself could be both, no?